src/include/txn.i

```diff
@@ -85,8 +85,9 @@ __wt_timestamp_set_zero(wt_timestamp_t *ts)
 
 #else /* WT_TIMESTAMP_SIZE != 8 */
 
-#define	WT_WITH_TIMESTAMP_READLOCK(s, l, e)	do {                    \
+#define	WT_WITH_TIMESTAMP_READLOCK(s, l, stat, e) do {                  \
 	__wt_readlock((s), (l));                                        \
+	WT_STAT_CONN_INCR((s), stat);                                   \
 	e;                                                              \
 	__wt_readunlock((s), (l));                                      \
 } while (0)
```

Comment:
I'd move the statistics set outside of the lock, there's no reason to do it inside the lock, is there?

Probably doesn't matter, but it feels odd.